application/aam-backend-service/src/main/kotlin/com/aamdigital/aambackendservice/skill/skilllab/SkillLabSyncUserProfileUseCase.kt

```diff
@@ -0,0 +1,105 @@
+package com.aamdigital.aambackendservice.skill.skilllab
+
+import com.aamdigital.aambackendservice.domain.UseCaseOutcome
+import com.aamdigital.aambackendservice.error.AamErrorCode
+import com.aamdigital.aambackendservice.skill.core.SyncUserProfileData
+import com.aamdigital.aambackendservice.skill.core.SyncUserProfileRequest
+import com.aamdigital.aambackendservice.skill.core.SyncUserProfileUseCase
+import com.aamdigital.aambackendservice.skill.domain.EscoSkill
+import com.aamdigital.aambackendservice.skill.domain.SkillUsage
+import com.aamdigital.aambackendservice.skill.domain.UserProfile
+import com.aamdigital.aambackendservice.skill.repository.SkillLabUserProfileEntity
+import com.aamdigital.aambackendservice.skill.repository.SkillLabUserProfileRepository
+import com.aamdigital.aambackendservice.skill.repository.SkillReferenceEntity
+
+enum class SkillLabSyncUserProfileErrorCode : AamErrorCode {
+    IO_ERROR
+}
+
+class SkillLabSyncUserProfileUseCase(
+    private val skillLabClient: SkillLabClient,
+    private val skillLabUserProfileRepository: SkillLabUserProfileRepository,
+) : SyncUserProfileUseCase() {
+
+    override fun apply(request: SyncUserProfileRequest): UseCaseOutcome<SyncUserProfileData> {
+        val userProfile = skillLabClient.fetchUserProfile(
+            externalIdentifier = request.userProfile
+        )
+
+        val allSkillsEntities = getSkillEntities(userProfile.profile)
+        val userProfileEntity = fetchUserProfileEntity(userProfile.profile, allSkillsEntities)
+
+        if (!userProfileEntity.mobileNumber.isNullOrBlank()) {
+            userProfileEntity.mobileNumber = userProfileEntity.mobileNumber
+                ?.replace(" ", "")
+                ?.replace("-", "")
+                ?.trim()
```

Comment:
shall we keep the original property as well and store transformed/cleaned in a separate field, in case that is useful for debugging etc.?